libsql-sqlite3/src/wal.c

```diff
@@ -4346,6 +4347,40 @@ int sqlite3DestroyCreateWal(void *self) {
     return SQLITE_OK;
 }
 
+int make_ref_counted_create_wal(libsql_create_wal create_wal, RefCountCreateWal **out) {
+    RefCountCreateWal *p = (RefCountCreateWal*)sqlite3MallocZero(sizeof(RefCountCreateWal));
+    if (!p) return SQLITE_NOMEM;
+    p->n = 1;
+    p->ref = create_wal;
+    *out = p;
+    return SQLITE_OK;
+}
+
+/*
+ * Decrease the ref count and call the create_wal destructor when the count reaches 0.
+ * Must be called from withing a critical section.
+ */
+void destroy_create_wal(RefCountCreateWal **p) {
```

Comment:
There's no reason for this function to take `RefCountCreateWal **`, it could just take `RefCountCreateWal *`, you dereference all the occurrences anyway. And it will make it easier to reason about what this parameter actually is - a pointer to create_wal, not a pointer-to-pointer-to-create_wal